### ISSUE TITLE
[Request] Set preset temperatures at run time

### ISSUE THREAD

**Issue by ScratMan** [ScratMan/HASmartThermostat#17]
### Discussed in https://github.com/ScratMan/HASmartThermostat/discussions/14

<div type='discussions-op-text'>

<sup>Originally posted by **nagyrobi** November 17, 2021</sup>
One of the problems of HA's built-in `generic_thermostat` is that the temperature for `away` mode is hardcoded in the configuration, it cannot be adjusted at run time.

Thankfully `smart_thermostat` supports `away`, `eco`, `boost`, `comfort`, `home`, `sleep`, `activity` each with its own temperature value configurable.

It would be nice to have the possibility to adjust these during runtime, and use HA's history database to always restore to the last known value after restart.

</div>

**Comment by m4dm4rtig4n** [comment]
Personally, I am waiting for this feature to switch to Smart thermostat.
Without this mode, I have to review my management of the heating systems, which work with dynamic presets :/

**Comment by ScratMan** [comment]
Hello, please test the new [2021.11.9-beta1 release](https://github.com/ScratMan/HASmartThermostat/releases/tag/2021.11.9-beta1).
I added a service to set the PID gains directly from the UI (service call working both in YAML and UI mode).

**Comment by ScratMan** [comment]
Hello @m4dm4rtig4n, could you please give a try to [v2021.11.9-beta2](https://github.com/ScratMan/HASmartThermostat/releases/tag/2021.11.9-beta2) that includes a service to set the preset temperatures through Home Assistant UI ?
Thanks

**Comment by m4dm4rtig4n** [comment]
I have try but i have lost all current present :/

My config : 
```
- platform: smart_thermostat
#- platform: simple_thermostat
  name: {{ climate_ha }}
  unique_id: {{ climate_ha }}
  heater: {{ climate.heater }}
  target_sensor: {{climate.target_sensor}}
#  target_temp: 15
  min_temp: 7
  max_temp: 26
  ac_mode: false
#  cold_tolerance: 0.3
#  hot_tolerance: 0
  min_cycle_duration:
    seconds: 5
  keep_alive:
    minutes: 3
  initial_hvac_mode: "heat"
#  away_temp: 16
  precision: 0.1
  kp: 5
  ki: 0.01
  kd: 500
  pwm: 00:15:00
```

![image](https://user-images.githubusercontent.com/7872518/143793233-f722f5f8-d2f3-421f-bc17-3d0d18c1ff61.png)

**Comment by ScratMan** [comment]
Hi, you should try by either keeping the preset temperatures settings in the YAML, or try to use the service once to set the presets you want, and then reload the integration (use smart thermostat reload service).

**Comment by ScratMan** [comment]
Closed with [release v2021.11.9](https://github.com/ScratMan/HASmartThermostat/releases/tag/2021.11.9)
